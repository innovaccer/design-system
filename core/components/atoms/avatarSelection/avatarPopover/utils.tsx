import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
