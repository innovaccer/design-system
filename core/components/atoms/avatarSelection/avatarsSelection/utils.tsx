import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
