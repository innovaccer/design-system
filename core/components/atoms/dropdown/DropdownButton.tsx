import * as React from 'react';
import classNames from 'classnames';
import { Icon, Text } from '@/index';
import { IconType } from '@/common.type';
import dropdownButtonStyles from '@css/components/dropdownButton.module.css';
import buttonStyles from '@css/components/button.module.css';
import textStyles from '@css/components/text.module.css';

export type DropDownButtonSize = 'tiny' | 'regular';

export interface TriggerProps {
  /**
   * Size of `Dropdown` trigger button
   * @default "regular"
   */
  triggerSize?: DropDownButtonSize;
  /**
   * Material icon name
   */
  icon?: string;
  /**
   * Set type of Icon
   */
  iconType?: IconType;
  /**
   * String to show inside `Dropdown trigger` when no options are selected
   * @default "Select"
   */
  placeholder?: string;
  /**
   * Label inside `Dropdown trigger button`
   */
  inlineLabel?: string;
  /**
   * Determines if `Dropdown` is disabled
   */
  disabled?: boolean;
  /**
   * Determines if type of `dropdown` is a menu
   * @default false
   */
  menu?: boolean;
  /**
   * Determines if `Dropdown` has error
   */
  error?: boolean;
}

export interface DropdownButtonProps extends TriggerProps {
  children?: React.ReactText;
  open?: boolean;
}

const DropdownButton = React.forwardRef<HTMLButtonElement, DropdownButtonProps>((props, ref) => {
  const {
    triggerSize = 'regular',
    placeholder = 'Select',
    menu = false,
    children,
    icon,
    disabled,
    open,
    inlineLabel,
    error,
    iconType,
    ...rest
  } = props;

  const buttonDisabled = disabled ? 'disabled' : 'default';
  const trimmedPlaceholder = placeholder.trim();
  const value = children ? children : trimmedPlaceholder;
  const iconName = !menu ? 'keyboard_arrow_down' : icon ? icon : 'more_horiz';

  const buttonClass = classNames({
    [buttonStyles['Button']]: true,
    [dropdownButtonStyles.DropdownButton]: true,
    [dropdownButtonStyles[`DropdownButton--${triggerSize}`]]: triggerSize,
    [dropdownButtonStyles[`DropdownButton--${triggerSize}Square`]]: menu,
    [dropdownButtonStyles['DropdownButton--placeholder']]: !children && !menu,
    [dropdownButtonStyles['DropdownButton--icon']]: icon,
    [dropdownButtonStyles['DropdownButton--open']]: open,
    [dropdownButtonStyles['DropdownButton--error']]: error,
  });

  const textClass = classNames({
    [textStyles['Text']]: true,
    [textStyles['Text--regular']]: true,
    [dropdownButtonStyles['DropdownButton-text']]: true,
  });

  return (
    <button
      ref={ref}
      type="button"
      value={children}
      className={buttonClass}
      disabled={disabled}
      tabIndex={0}
      data-test="DesignSystem-DropdownTrigger"
      {...rest}
    >
      {!menu && (
        <div className={dropdownButtonStyles['DropdownButton-wrapper']}>
          {inlineLabel && (
            <Text appearance="subtle" className="mr-4 white-space-nowrap">
              {`${inlineLabel.trim().charAt(0).toUpperCase()}${inlineLabel.trim().slice(1)}`}
            </Text>
          )}
          {icon && !inlineLabel && (
            <Icon appearance={buttonDisabled} className="d-flex align-items-center mr-4" name={icon} type={iconType} />
          )}
          {value && <span className={textClass}>{value}</span>}
        </div>
      )}
      <Icon appearance={buttonDisabled} name={iconName} type={iconType} />
    </button>
  );
});

DropdownButton.displayName = 'DropdownButton';

export default DropdownButton;
