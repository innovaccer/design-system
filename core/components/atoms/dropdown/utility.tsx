import { OptionSchema as Option } from './option';

export const getSearchedOptions = (options: any, searchTerm: string) => {
  const result = options.filter((option: Option) => option.label.toLowerCase().includes(searchTerm.toLowerCase()));
  return result;
};

export const _isEqual = (arr1: Option[], arr2: Option[]) =>
  arr1.length === arr2.length &&
  arr1.every((option, index) => option.value === arr2[index].value || option.label === arr2[index].label);

export const _isControlled = (selected?: Option[]) => selected !== undefined;

export const _isOpenControlled = (open?: boolean) => open !== undefined;

export const _showSelectedItems = (bulk: boolean, searchTerm: string, withCheckbox?: boolean) =>
  bulk && withCheckbox && searchTerm === '';

export const _isSelectAllPresent = (
  searchTerm: string,
  bulkOptions: number,
  withSelectAll: boolean,
  withCheckbox?: boolean
) => withCheckbox && withSelectAll && bulkOptions === 0 && searchTerm === '';

export const scrollTo = (element: Element, top: number) => {
  element.scrollTo(0, top);
};

export const scrollIntoView = (menuElement: HTMLDivElement | null, focusedElement: HTMLElement) => {
  const menuRect = menuElement?.getBoundingClientRect();
  const focusedRect = focusedElement.getBoundingClientRect();
  const overscroll = focusedElement.offsetHeight;

  if (focusedRect.bottom > menuRect!.bottom && menuElement) {
    scrollTo(menuElement, focusedElement.offsetTop - menuRect!.height + overscroll);
  } else if (focusedRect.top < menuRect!.top && menuElement) {
    scrollTo(menuElement, focusedElement.offsetTop - overscroll);
  }
};

export const getSelectAll = (selected: Option[], optionsLength: number, disabledOptionsLength: number) => {
  if (selected.length) {
    if (selected.length > 0 && disabledOptionsLength > 0 && selected.length === optionsLength - disabledOptionsLength) {
      return { indeterminate: true, checked: true }; //
    }
    const indeterminate = selected.length > 0 && selected.length !== optionsLength;
    const checked = selected.length > 0 && selected.length === optionsLength;
    const obj = { checked, indeterminate };
    return obj;
  }
  return { indeterminate: false, checked: false };
};
