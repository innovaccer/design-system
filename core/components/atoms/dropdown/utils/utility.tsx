const getSearchedOptions = (options: any, searchTerm: string) => {
  const result = options.filter((option: any) => option.label.toLowerCase().includes(searchTerm.toLowerCase()));
  return result;
};

export const getOptions = (offset: number, limit: number, searchTerm: string, options: any) => {
  const searchedOptions = searchTerm ? getSearchedOptions(options, searchTerm) : options;
  return new Promise(resolve => {
    resolve({
      offset,
      slicedOptions: searchedOptions.slice(offset, offset + limit),
      options: searchedOptions,
    });
  });
};

export const getValuesFromSelectedObj = (selectedArray: any[] = []) => {
  const selectedValues: any[] = [];

  selectedArray.forEach((selectedObj: any) => {
    const { value } = selectedObj;
    selectedValues.push(value);
  });

  return selectedValues;
};

export const getIndexesFromSelectedObj = (selectedArray: any[] = []) => {
  return (Array.from(Array(selectedArray.length).keys()));
};

export const getValuesFromInd = (selectedArray: number[], options: any[]) => {
  const result: any[] = [];
  selectedArray.forEach(ind => {
    result.push(options[ind].value);
  });

  return result;
};

export const getLabelsFromInd = (selectedArray: number[], options: any[]) => {
  const result: string[] = [];
  selectedArray.forEach(ind => {
    result.push(options[ind].label);
  });

  return result;
};
