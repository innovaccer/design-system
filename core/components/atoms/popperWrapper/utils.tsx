// Helper function to convert CSS class names to style objects
export const createStyleFromClass = (className: string) => {
  if (!className) return {};

  // Create a temporary element to get computed styles
  const tempElement = document.createElement('div');
  tempElement.className = className;
  document.body.appendChild(tempElement);

  const computedStyle = window.getComputedStyle(tempElement);
  const style: React.CSSProperties = {};

  // Extract relevant animation properties
  const animation = computedStyle.animation;
  const animationDuration = computedStyle.animationDuration;
  const animationTimingFunction = computedStyle.animationTimingFunction;
  const animationFillMode = computedStyle.animationFillMode;
  const opacity = computedStyle.opacity;
  const transform = computedStyle.transform;

  if (animation && animation !== 'none') {
    style.animation = animation;
  }
  if (animationDuration && animationDuration !== '0s') {
    style.animationDuration = animationDuration;
  }
  if (animationTimingFunction && animationTimingFunction !== 'ease') {
    style.animationTimingFunction = animationTimingFunction;
  }
  if (animationFillMode && animationFillMode !== 'none') {
    style.animationFillMode = animationFillMode;
  }
  if (opacity && opacity !== '1') {
    style.opacity = opacity;
  }
  if (transform && transform !== 'none') {
    style.transform = transform;
  }

  document.body.removeChild(tempElement);
  return style;
};
