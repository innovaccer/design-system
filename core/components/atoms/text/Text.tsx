import * as React from 'react';
import GenericText from '../_text';
import classNames from 'classnames';
import { BaseHtmlProps, BaseProps } from '@/utils/types';
import { TextColor } from '@/common.type';
import styles from '@css/components/text.module.css';

export type TextSize = 'small' | 'regular' | 'large';
export type TextAppearance = 'default' | 'white' | 'destructive' | 'subtle' | 'disabled' | 'success' | 'link';

export interface TextProps extends BaseProps, BaseHtmlProps<HTMLSpanElement> {
  /**
   * Text to be rendered
   */
  children: React.ReactText;
  /**
   * Denotes weight of `Text`
   */
  weight?: 'strong' | 'medium';
  /**
   * Changes size of `Text`**[SOON\_TO\_BE\_DEPRECATED]**
   */
  small?: boolean;
  /**
   * State of `Text`
   */
  appearance?: TextAppearance;
  /**
   * Size of `Text`
   */
  size?: TextSize;
  /**
   * Color of `Text`
   */
  color?: TextColor;
}

export const Text = React.forwardRef<HTMLSpanElement, TextProps>((props, ref) => {
  const { appearance = 'default', size = 'regular', children, weight, small, className, color, ...rest } = props;

  const classes = classNames(
    {
      [styles.Text]: true,
      [styles[`Text--${appearance}`]]: !color && appearance,
      [styles[`Text--${weight}`]]: weight,
      [styles[`Text--${size}`]]: size,
      [`color-${color}`]: color,
      /* SOON_TO_BE_DEPRECATED */
      [styles['Text--small']]: size === 'small' || small,
    },
    className
  );

  return (
    <GenericText ref={ref} data-test="DesignSystem-Text" {...rest} className={classes} componentType="span">
      {children}
    </GenericText>
  );
});

Text.displayName = 'Text';

Text.defaultProps = {
  appearance: 'default',
  size: 'regular',
};

export default Text;
