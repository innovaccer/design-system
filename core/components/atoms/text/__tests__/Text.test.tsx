import * as React from 'react';
import Text, { TextProps as Props, Appearance, Size } from '../Text';
import { render } from '@testing-library/react';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';
import { _isEqual } from '../../dropdown/utility';

const sizes: Size[] = ['small', 'regular', 'large'];
const appearances: Appearance[] = ['default', 'white', 'destructive', 'disabled', 'subtle', 'success', 'link'];
const weight = ['strong', 'medium'];
const BooleanValue = [true, false];
const StringValue = 'Sample String';

describe('Text component', () => {
  const mapper = {
    children: valueHelper(StringValue, { required: true }),
    size: valueHelper(sizes, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Text {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Text component', () => {
  const mapper = {
    children: valueHelper(StringValue, { required: true }),
    appearance: valueHelper(appearances, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Text {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Text component', () => {
  const mapper = {
    children: valueHelper(StringValue, { required: true }),
    weight: valueHelper(weight, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Text {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Text component', () => {
  const mapper = {
    children: valueHelper(StringValue, { required: true }),
    small: valueHelper(BooleanValue, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = render(<Text {...attr} />);
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Text component', () => {
  it('should render the text', () => {
    const { getByTestId } = render(<Text>{'Design System'}</Text>);
    expect(getByTestId('DesignSystem-Text')).toHaveTextContent('Design System');
  });

  it('should render componentType="span"', () => {
    const { getByTestId } = render(<Text>{'Design System'}</Text>);
    expect(getByTestId('DesignSystem-Text').tagName).toEqual('SPAN');
  });

  describe('Text Component with overwrite class', () => {
    it('overwrite Text class', () => {
      const { getByTestId } = render(<Text className="TextClass">{'Text'}</Text>);
      expect(getByTestId('DesignSystem-Text')).toHaveClass('TextClass');
    });
  });

  describe('Text Component with Prop:weight', () => {
    it('should have the class Text--strong when weight="strong"', () => {
      const { getByTestId } = render(<Text weight="strong">{'Design System'}</Text>);
      expect(getByTestId('DesignSystem-Text')).toHaveClass('Text--strong');
    });

    it('should have the class Text--medium when weight="medium"', () => {
      const { getByTestId } = render(<Text weight="medium">{'Design System'}</Text>);
      expect(getByTestId('DesignSystem-Text')).toHaveClass('Text--medium');
    });
  });

  describe('Text Component with Prop:small', () => {
    it('should have the class Text--small when small="true"', () => {
      const { getByTestId } = render(<Text small={true}>{'Design System'}</Text>);
      expect(getByTestId('DesignSystem-Text')).toHaveClass('Text--small');
    });
  });

  describe('Text Component with Prop:appearances', () => {
    it('should have the class Text--default when appearances="default"', () => {
      const { getByTestId } = render(<Text appearance="default">{'Design System'}</Text>);
      expect(getByTestId('DesignSystem-Text')).toHaveClass('Text--default');
    });

    appearances.forEach((appearance) => {
      it(`should have the Text--${appearance} class when appearances=${appearance} `, () => {
        const { getByTestId } = render(<Text appearance={appearance}>{'Design System'}</Text>);
        expect(getByTestId('DesignSystem-Text')).toHaveClass(`Text--${appearance}`);
      });
    });
  });
});
