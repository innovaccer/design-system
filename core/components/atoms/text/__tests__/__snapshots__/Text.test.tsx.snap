// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component 
  children: "Sample String", appearance: "default"
 1`] = `
<body>
  <div>
    <span
      class="Text Text--default Text--regular"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>
</body>
`;

exports[`Text component 
  children: "Sample String", appearance: "destructive"
 1`] = `
<body>
  <div>
    <span
      class="Text Text--destructive Text--regular"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>
</body>
`;

exports[`Text component 
  children: "Sample String", appearance: "disabled"
 1`] = `
<body>
  <div>
    <span
      class="Text Text--disabled Text--regular"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>
</body>
`;

exports[`Text component 
  children: "Sample String", appearance: "link"
 1`] = `
<body>
  <div>
    <span
      class="Text Text--link Text--regular"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>
</body>
`;

exports[`Text component 
  children: "Sample String", appearance: "subtle"
 1`] = `
<body>
  <div>
    <span
      class="Text Text--subtle Text--regular"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>
</body>
`;

exports[`Text component 
  children: "Sample String", appearance: "success"
 1`] = `
<body>
  <div>
    <span
      class="Text Text--success Text--regular"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>
</body>
`;

exports[`Text component 
  children: "Sample String", appearance: "white"
 1`] = `
<body>
  <div>
    <span
      class="Text Text--white Text--regular"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>
</body>
`;

exports[`Text component 
  children: "Sample String", size: "large"
 1`] = `
<body>
  <div>
    <span
      class="Text Text--default Text--large"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>
</body>
`;

exports[`Text component 
  children: "Sample String", size: "regular"
 1`] = `
<body>
  <div>
    <span
      class="Text Text--default Text--regular"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>
</body>
`;

exports[`Text component 
  children: "Sample String", size: "small"
 1`] = `
<body>
  <div>
    <span
      class="Text Text--default Text--small"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>
</body>
`;

exports[`Text component 
  children: "Sample String", small: false
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="Text Text--default Text--regular"
        data-test="DesignSystem-Text"
      >
        Sample String
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="Text Text--default Text--regular"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Text component 
  children: "Sample String", small: true
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="Text Text--default Text--regular Text--small"
        data-test="DesignSystem-Text"
      >
        Sample String
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="Text Text--default Text--regular Text--small"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Text component 
  children: "Sample String", weight: "medium"
 1`] = `
<body>
  <div>
    <span
      class="Text Text--default Text--medium Text--regular"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>
</body>
`;

exports[`Text component 
  children: "Sample String", weight: "strong"
 1`] = `
<body>
  <div>
    <span
      class="Text Text--default Text--strong Text--regular"
      data-test="DesignSystem-Text"
    >
      Sample String
    </span>
  </div>
</body>
`;
